app/main/services/query_builder.py

```diff
@@ -69,8 +74,11 @@ def highlight_clause(mapping):
 
 
 def is_filtered(mapping, query_args):
-    return len(set(query_args.keys()).intersection(
-        ["filter_" + field for field in mapping.filter_fields_set])) > 0
+    return bool(frozenset(
+        maybe_name[0]
+        for prefix, *maybe_name in (arg_key.split("_", 1) for arg_key in query_args.keys())
+        if prefix == "filter" and maybe_name  # maybe_name could be an empty seq if no underscores were found
```

Comment:
The arg_key would have to just be 'filter' in the case that `maybe_name` was an empty sequence?

Reply:
Guess so, yes.

Reply:
We're never going to have a field called 'filter' though, right?

Reply:
Robust by default. This is where we process query strings, it would be totally possible for a client to make a request `&filter_=blah` and we don't want to do something unpredictable in that case, rather ignore it.